File: test_archives/py_tests/batch4/test_function4.py
from functions.function4 import function4
import pytest

def test_function4_1():
    assert function4(6, "Hello", [2, 5, 8]) == 54.0

def test_function4_2():
    assert function4(9, "World", [1, 3, 4]) == 66.0

File: test_archives/cpp_tests/batch4/test_function4.cpp
#include "catch.hpp"
#include "../functions/function4.cpp"

TEST_CASE("Test function4 number 1", "[function4]") {
    REQUIRE(function4(6, "Hello", {2, 5, 8}) == 54.0);
}

TEST_CASE("Test function4 number 2", "[function4]") {
    REQUIRE(function4(9, "World", {1, 3, 4}) == 66.0);
}